### PR TITLE
CHANGELOG for v0.6.0

### PR DESCRIPTION
**Pull request by alexcb**
Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,131 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
...
+- [`LOCALLY`](https://docs.earthly.dev/docs/earthfile#locally) has been promoted to GA [#580](https://github.com/earthly/earthly/issues/580).
+- [`RUN --interactive` and `RUN --interactive-keep`](https://docs.earthly.dev/docs/earthfile#run) have been promoted to GA [#693](https://github.com/earthly/earthly/issues/693).
+- [`IF`](https://docs.earthly.dev/docs/earthfile#if) and [`FOR`](https://docs.earthly.dev/docs/earthfile#for) have been promoted to GA [#779](https://github.com/earthly/earthly/issues/779).
+- If a `SAVE ARTIFACT` is unsafe (writing to a directory outside of the Earthfile directory), it'll require the `--force` flag.
+- `.earthlyignore` no longer includes any implicit entries like `Earthfile` or `.earthlyignore`. These will need to be specified explicitly. [#1294](https://github.com/earthly/earthly/issues/1294)
+- The console output now has an improved structure [#1226](https://github.com/earthly/earthly/pull/1226).
+- Fixed homebrew installation on macOS 12. [#1370](https://github.com/earthly/earthly/pull/1370), [homebrew/earthly#13](https://github.com/earthly/homebrew-earthly/pull/13)
+### Changed
```
Comment (vladaionescu) [inline CHANGELOG.md:51]: I noticed people don't immediately notice these key breaking changes when they skim the changelog. Shall we move the `Changed` section above the `Added` section?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,131 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
...
+
+  Earthly v0.6 (new way)
+
+  ```
+  earthly +some-target --NAME=john
+  ```
+
+  This change is part of the [UDC proposal #581](https://github.com/earthly/earthly/issues/581). The old way of passing args is deprecated and will be removed in a future version (however, it still works in 0.6).
+<!--changelog-parser-ignore-end-->
+- Add builtin args `USERPLATFORM`, `USEROS`, `USERARCH`, and `USERVARIANT` which represent the platform, OS, architecture, and processor variant of the system Earthly is being called from [#1251](https://github.com/earthly/earthly/pull/1251). Thanks to @akrantz01 for the contribution!
```
Comment (vladaionescu) [inline CHANGELOG.md:115]: This should go under the `Added` section

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,131 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
...
+- Support for required ARGs (`ARG --required foo`) [#904](https://github.com/earthly/earthly/issues/904). Thanks to @camerondurham for the contribution!
```
Comment (vladaionescu) [inline CHANGELOG.md:116]: Same with this one

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,131 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
...
+- Add a config item for buildkit's `max_parallelism` configuration. Use this to increase parallelism for faster builds or decrease parallelism when resources are constraint. The default is 20. [#1308](https://github.com/earthly/earthly/issues/1308)
```
Comment (vladaionescu) [inline CHANGELOG.md:117]: Same with this one

File: CHANGELOG.md (hunk 1 above)
Comment (alexcb) [reply CHANGELOG.md:51]: That's a good point! I'll move this up.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**